### PR TITLE
feat(samples/pytorch): Adding compilation test for PyTorch samples KFP

### PR DESCRIPTION
**Pull request by ankan94**
**Description of your changes:**
Adding unit tests for compilation of the PyTorch sample yamls .

**Checklist:**
- [X] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

**Comment by ankan94** [comment]
@googlebot I consent.

**Comment by shrinath-suresh** [comment]
@googlebot I consent

File: samples/contrib/pytorch-samples/test_compile_yamls.py [hunk 1]
```diff
@@ -0,0 +1,424 @@
...
+PRED_OP = load_component_from_file("./common/prediction/component.yaml")
+
+
+class ComponentCompileTest(unittest.TestCase):  #pylint: disable=too-many-instance-attributes
+    """Test cases for compilation of yamls."""
+
+    def setUp(self):
+        """Set up."""
+        super(ComponentCompileTest, self).setUp()
+        self.input_request = "https://kubeflow-dataset.s3.us-east-2.amazonaws.com/" \
```
Comment (chauhang) [inline samples/contrib/pytorch-samples/test_compile_yamls.py:53]: Please use local file or github loction

File: samples/contrib/pytorch-samples/test_compile_yamls.py [hunk 2]
```diff
@@ -0,0 +1,424 @@
...
+                    host_name=self.isvc_name,
+                    input_request=self.input_request,
+                    cookie=self.cookie,
+                    url=self.ingress_gateway,
+                    model=self.model_name,
+                    inference_type="explain",
+                ).after(pred_task).set_display_name("Explanation")
+            )
+
+        compiler.Compiler().compile(
```
Comment (chauhang) [inline samples/contrib/pytorch-samples/test_compile_yamls.py:262]: How are the failure cases being caught?

File: samples/contrib/pytorch-samples/test_compile_yamls.py [hunk 3]
```diff
@@ -0,0 +1,424 @@
...
+                    cloud.google.com/gke-accelerator: {}
+        """.format(deploy, namespace, model_uri, gpu_count, accelerator)
+            # Update inferenceservice_yaml for GPU inference
+            deploy_task = (
+                COMPONENT_DEPLOY(
+                    action="apply", inferenceservice_yaml=isvc_yaml
+                ).after(minio_mar_upload).set_display_name("Deployer")
+            )
+
+        compiler.Compiler().compile(
```
Comment (chauhang) [inline samples/contrib/pytorch-samples/test_compile_yamls.py:422]: How are the failure cases being caught?

**Review by chauhang — CHANGES_REQUESTED** [review]
Please see comments inline

File: samples/contrib/pytorch-samples/test_compile_yamls.py (hunk 3 above)
Comment (ankan94) [reply samples/contrib/pytorch-samples/test_compile_yamls.py:422]: @chauhang The google cloud pipeline compilation test which I referred to didn't include any negative cases. I'll try to add negative cases for this.

File: samples/contrib/pytorch-samples/test_compile_yamls.py (hunk 1 above)
Comment (ankan94) [reply samples/contrib/pytorch-samples/test_compile_yamls.py]: Done.

File: samples/contrib/pytorch-samples/test_compile_yamls.py (hunk 2 above)
Comment (ankan94) [reply samples/contrib/pytorch-samples/test_compile_yamls.py:262]: Added the two failure cases

**Comment by ankan94** [comment]
@googlebot I consent.

**Comment by shrinath-suresh** [comment]
@googlebot I consent.

**Comment by chauhang** [comment]
@googlebot I consent

**Comment by jagadeeshi2i** [comment]
@googlebot I consent

**Review by shrinath-suresh — APPROVED** [review]

**Pull request merged**